ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.js

```diff
@@ -118,9 +124,9 @@ const ConfirmAddSuggestedToken = () => {
 
   const handleAddTokensClick = useCallback(async () => {
     await Promise.all(
-      suggestedAssets.map(async ({ asset, id }) => {
-        await dispatch(acceptWatchAsset(id));
-
+      suggestedAssets.flatMap(({ asset, id }) => [
+        dispatch(resolvePendingApproval(id, null)),
+        dispatch(acceptWatchAsset(id)),
```

Comment:
This will not execute if I just add another line with `await dispatch(acceptWatchAsset(id));`, I have to return the whole thing with a flatMap.

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/45107/workflows/70f1b4d8-b672-4e04-984c-f22e56e3acad/jobs/1275053

Reply:
Nice one!

Reply:
I know this code won't last long but would another option have been to dispatch the `acceptWatchAsset` with a `then` to chain it to the `resolvePendingApproval`?

It's probably redundant as there was no error handling here anyway but it would ensure we don't resolve the approval and hide the confirmation until the accept side effect is completed.

Reply:
It turns out that the issue was that with the event handler being async, it was trying to assert things before they were being called. It's been sorted using `act` (https://legacy.reactjs.org/docs/testing-recipes.html#act).